customer-service/src/main/java/io/eventuate/examples/tram/ordersandcustomers/customers/service/CustomerService.java

```diff
@@ -0,0 +1,115 @@
+package io.eventuate.examples.tram.ordersandcustomers.customers.service;
+
+import io.eventuate.examples.tram.ordersandcustomers.common.domain.Money;
+import io.eventuate.examples.tram.ordersandcustomers.customers.domain.Customer;
+import io.eventuate.examples.tram.ordersandcustomers.customers.domain.CustomerRepository;
+import io.eventuate.examples.tram.ordersandcustomers.customers.domain.ReservedCredit;
+import io.eventuate.examples.tram.ordersandcustomers.customers.domain.ReservedCreditRepository;
+import io.eventuate.examples.tram.ordersandcustomers.customers.domain.events.CustomerCreditReservationFailedEvent;
+import io.eventuate.examples.tram.ordersandcustomers.customers.domain.events.CustomerCreditReservedEvent;
+import io.eventuate.examples.tram.ordersandcustomers.customers.domain.events.CustomerValidationFailedEvent;
+import io.eventuate.tram.events.publisher.ResultWithEvents;
+import io.eventuate.tram.messaging.common.Message;
+import io.eventuate.tram.spring.events.publisher.ReactiveDomainEventPublisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.transaction.reactive.TransactionalOperator;
+import reactor.core.publisher.Mono;
+
+import java.math.BigDecimal;
+import java.util.Collections;
+import java.util.List;
+
+public class CustomerService {
+
+  private Logger logger = LoggerFactory.getLogger(getClass());
+
+  private CustomerRepository customerRepository;
+
+  private ReservedCreditRepository reservedCreditRepository;
+
+  private ReactiveDomainEventPublisher domainEventPublisher;
+
+  private TransactionalOperator transactionalOperator;
+
+  public CustomerService(CustomerRepository customerRepository,
+                         ReservedCreditRepository reservedCreditRepository,
+                         ReactiveDomainEventPublisher domainEventPublisher,
+                         TransactionalOperator transactionalOperator) {
+    this.customerRepository = customerRepository;
+    this.reservedCreditRepository = reservedCreditRepository;
+    this.domainEventPublisher = domainEventPublisher;
+    this.transactionalOperator = transactionalOperator;
+  }
+
+  public Mono<Customer> createCustomer(String name, Money creditLimit) {
+    ResultWithEvents<Customer> customerWithEvents = Customer.create(name, creditLimit);
+
+    return customerRepository
+            .save(customerWithEvents.result)
+            .flatMap(customer -> domainEventPublisher.publish(Customer.class, customer.getId(), customerWithEvents.events).collectList().map(messages -> customer))
+            .as(transactionalOperator::transactional);
```

Comment:
The code should use `@Transactional`

Reply:
Chris, as we discussed there are problems with that annotation, and I thought that we decided to use transnational operator

Reply:
The docs say that @transactional, which is the standard spring mechanism, should work for reactive.

Therefore, if it's not working in the example application then either the application is incorrect or there is a bug in spring.

Reply:
ok, I will research